File: product-service/src/main/java/com/intexsoft/webshop/productservice/dto/attribute/AttributeUpdateDto.java
package com.intexsoft.webshop.productservice.dto.attribute;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;
import jakarta.annotation.Nullable;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Positive;
import jakarta.validation.constraints.Size;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.FieldDefaults;

@Data
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
@JsonIgnoreProperties(ignoreUnknown = true)
@JsonInclude(JsonInclude.Include.NON_NULL)
public class AttributeUpdateDto {
    @Nullable
    @Positive
    Long id;
    @NotBlank
    @Size(max = 50)
    String label;
}

File: product-service-kt/src/main/kotlin/com/intexsoft/webshop/productservicekt/dto/attribute/AttributeUpdateDto.kt
package com.intexsoft.webshop.productservicekt.dto.attribute

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Positive
import jakarta.validation.constraints.Size

@JsonIgnoreProperties(ignoreUnknown = true)
data class AttributeUpdateDto(
    @field:Positive
    val id: Long?,
    @field:NotBlank
    @field:Size(max = 50)
    val label: String?
)